api/graphite.go

```diff
@@ -293,22 +293,13 @@ func (s *Server) renderMetrics(ctx *middleware.Context, request models.GraphiteR
 	}
 	plan, err := expr.NewPlan(exprs, fromUnix, toUnix, mdp, stable, opts)
 	if err != nil {
-		fun, isUnknownFunction := err.(expr.ErrUnknownFunction)
-		err := response.WrapError(err)
+		unknownFunction := expr.ErrUnknownFunction("")
+		isUnknownFunction := errors.As(err, &unknownFunction)
 		if isUnknownFunction {
-			if request.NoProxy {
-				// note this should never happen:
-				// When graphite issues a request (and sets the request.NoProxy flag),
-				// it should be for raw data only without any function processing.
-				ctx.Error(err.HTTPStatusCode(), err.Error())
-				return
```

Comment:
why did you take out this clause? if this condition is triggered it would result in an infinite proxy loop.

Reply:
`!request.NoProxy` is still a condition to proxy the request below. All the logic for unknown function is the same except for the stats call, so I just left the stats call alone.

Reply:
ah i see. you are right! nice code simplification!